### ISSUE TITLE
block parameter (like each) struggles to recognize type of assignments

### ISSUE THREAD

**Issue by dezza** [castwide/solargraph#558]
It seems like blocks prevent type from being recognized later on assignments, even when documented with yard tag. Without the `each` block it seems to figure it out.

```rb

def foo
  # @type [File]
  fs = File.open("foo", 'r')
  p fs # keyword lookup and completion works
end

def test
  # @param f [String]
  ['file1', 'file2.com'].each do |f|
    # @type [File]
    rs = File.open(f, 'r')
    p rs #fails keyword lookup and completion
  end
end
```

I tried to look over previous issues and https://solargraph.org/guides/yard and https://solargraph.org/guides/type-checking articles and couldn't find a mention of why this would be the case.

#374 Not supported?

**Comment by castwide** [comment]
This bug is present in 0.44.3, but in the current master, lookup and completion work for both `fs` and `rs`. The fix will be released in the next version, which I expect to publish in the next day or so.

**Comment by castwide** [comment]
Released in v0.45.0.
